aws-android-sdk-pinpoint/src/main/java/com/amazonaws/mobileconnectors/pinpoint/targeting/notification/PinpointNotificationActivity.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2016-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ * http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amazonaws.mobileconnectors.pinpoint.targeting.notification;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.content.pm.PackageManager;
+import android.os.Bundle;
+
+/**
+ * The Amazon Pinpoint push notification tracking/routing activity.
+ * This class is an Activity instead of BroadcastReceiver due to Android 12 restrictions around
+ * notification trampolining.
+ */
+public class PinpointNotificationActivity extends Activity {
+    private static String TAG = PinpointNotificationActivity.class.getSimpleName();
+
+    private static volatile NotificationClient notificationClient = null;
+
+    public static void setNotificationClient(NotificationClient notificationClient) {
+        PinpointNotificationActivity.notificationClient = notificationClient;
+    }
+
+    public static void setNotificationClient(NotificationClientBase notificationClientBase) {
```

Comment:
Can we avoid adding static methods in the activity? if the activity is recreated then this property will always be null.